mayastor/tests/nexus_multipath.rs

```diff
@@ -0,0 +1,173 @@
+    let test = Builder::new()
+        .name("nexus_shared_replica_test")
+        .network("10.1.0.0/16")
+        .add_container("ms1")
+        .with_clean(true)
+        .build()
+        .await
+        .unwrap();
+
+    let mut hdls = test.grpc_handles().await.unwrap();
+
+    // create a pool on remote node
+    hdls[0]
+        .mayastor
+        .create_pool(CreatePoolRequest {
+            name: POOL_NAME.to_string(),
+            disks: vec!["malloc:///disk0?size_mb=64".into()],
+        })
+        .await
+        .unwrap();
+
+    // create replica, not shared
+    hdls[0]
+        .mayastor
+        .create_replica(CreateReplicaRequest {
+            uuid: UUID.to_string(),
+            pool: POOL_NAME.to_string(),
+            size: 32 * 1024 * 1024,
+            thin: false,
+            share: 0,
+        })
+        .await
+        .unwrap();
+
+    // create nexus on remote node with local replica as child
+    hdls[0]
+        .mayastor
+        .create_nexus(CreateNexusRequest {
+            uuid: UUID.to_string(),
+            size: 32 * 1024 * 1024,
+            children: [format!("loopback:///{}", UUID)].to_vec(),
+        })
+        .await
+        .unwrap();
+
+    // share replica
+    hdls[0]
+        .mayastor
+        .share_replica(ShareReplicaRequest {
+            uuid: UUID.to_string(),
+            share: 1,
+        })
+        .await
+        .unwrap();
+
+    let mayastor = MayastorTest::new(MayastorCliArgs::default());
+    let ip0 = hdls[0].endpoint.ip();
+    let nexus_name = format!("nexus-{}", UUID);
+    mayastor
+        .spawn(async move {
+            // create nexus on local node with remote replica as child
+            nexus_create(
+                &nexus_name,
+                32 * 1024 * 1024,
+                Some(UUID),
+                &[format!("nvmf://{}:8420/{}:{}", ip0, HOSTNQN, UUID)],
+            )
+            .await
+            .unwrap();
+            // publish nexus on local node over nvmf
+            nexus_lookup(&nexus_name)
+                .unwrap()
+                .share(ShareProtocolNexus::NexusNvmf, None)
+                .await
+                .unwrap();
+        })
+        .await;
+
+    // publish nexus on other node
+    hdls[0]
+        .mayastor
+        .publish_nexus(PublishNexusRequest {
+            uuid: UUID.to_string(),
+            key: "".to_string(),
+            share: ShareProtocolNexus::NexusNvmf as i32,
+        })
+        .await
+        .unwrap();
+
+    let nqn = format!("{}:nexus-{}", HOSTNQN, UUID);
+    let status = Command::new("nvme")
+        .args(&["connect"])
+        .args(&["-t", "tcp"])
+        .args(&["-a", "127.0.0.1"])
+        .args(&["-s", "8420"])
+        .args(&["-n", &nqn])
+        .status()
+        .unwrap();
+    assert!(
+        status.success(),
+        "failed to connect to local nexus, {}",
+        status
+    );
+
+    // The first attempt often fails with "Duplicate cntlid x with y" error from
```

Comment:
is this a kernel bug then?

Reply:
It's unclear. There are a few mentions of this error message in the wild but they involve users trying to use 2 NVMe SSDs of the same model and a firmware bug means they both present the same subnqn. In this case, having identical subnqns is necessary for ANA to work. It doesn't always happen, I just tried it manually on the CLI and the second connect worked just fine. I did try adding a delay in the cargo test and steadily increased it to 5s but that didn't seem to fix it.

Reply:
yeah, seems like it's confused somehow, guess this means our CSI plugin will have to do some kind of retry when connecting as well.